packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -154,6 +154,7 @@ class InspectorPreferencesController extends DisposableController
   static const _customPubRootDirectoriesStoragePrefix =
       'inspector.customPubRootDirectories';
   String? _mainScriptDir;
+  bool _checkedFlutterPubRoot = false;
```

Comment:
will `readCachedPubRootDirectories` ever need to be called more than once? What about when we disconnect and connect to a new app - in that case does `_checkedFlutterPubRoot` need to be reset?

Reply:
Ah good point! Resetting now on a new app connection (added a test for that as well)